zppy/utils.py

```diff
@@ -152,6 +152,18 @@ def getComponent(input_files):
     return component
 
 
+# -----------------------------------------------------------------------------
+
+
+def setMappingFile(c):
+    if c["mapping_file"] and (c["mapping_file"] != "glb"):
+        directory = os.path.dirname(c["mapping_file"])
+        if not directory:
+            machine_info = MachineInfo()
+            base_path = machine_info.config.get("diagnostics", "base_path")
+            c["mapping_file"] = os.path.join(base_path, "maps", c["mapping_file"])
```

Comment:
Or, @czender, are you planning to copy over the proper mapping files to a certain directory?

Reply:
@forsyth2 I also do not understand the difference between those two directories. They appear to duplicate eachother. To get things started, I copied the map mentioned above to the Mache-suggested location `/lcrc/group/e3sm/diagnostics/maps/` with 
`rsync -av /home/ac.zender/data/maps/map_ne30pg2_to_cmip6_180x360_aave.20200201.nc /lcrc/group/e3sm/diagnostics/maps/`
This is the first atmospheric map in that directory. @xylar and @chengzhuzhang is this where we want the atmospheric maps? If so, I can copy (many) others there, though maybe we should restrict the list to those accessible through `zppy` or `e2c` or...?

Reply:
@forsyth2 and @czender , a short answer now and a longer one tomorrow.

The e3sm/public_html/diagnostics directory is for diags. files that can be made public. There is an e3sm/private_diagnostics directory for data that must remain private to E3SM. e3sm/diagnostics combines the 2. Thus, you should put files in e3sm/public_html/diagnostics and get them from e3sm/diagnostics. Mache syncs the public and private diags. to the combined directory on LCRC and equivalents on other machines.

In retrospect or codes could probably have been made to look first in a public and then in a private directory on each machine for diagnostics but that isn't how they work at the moment, thus the redundant copy.

Reply:
A call like

mache sync diags from chrysalis

should take care of copying the files. But better yet ping me to do it when you're ready.

Reply:
Hi @czender and @forsyth2, my understanding is that we should write in data under /lcrc/group/e3sm/public_html/diagnostics/maps. And the other directory (/lcrc/group/e3sm/diagnostics/maps/) is where the users/tools supposed to read data from. 

I think it is okay to only keep a copy of files that are only needed by post-processing tools.

Reply:
Sounds good. However, I do not have write permission on `/lcrc/group/e3sm/public_html/diagnostics/maps`:
```
ac.zender@blueslogin4:/lcrc/group/e3sm/public_html/diagnostics/maps$ touch foo
touch: cannot touch ‘foo’: Permission denied
```
I am a member of E3SM group, and so I think @xylar could grant permission with `chmod 755 maps`:
```
ac.zender@blueslogin4:/lcrc/group/e3sm/public_html/diagnostics$ groups
cels collab lcrc E3SM
ac.zender@blueslogin4:/lcrc/group/e3sm/public_html/diagnostics$ ls -ld maps
drwxr-sr-x 2 ac.xasay-davis E3SM 4096 Oct 28 06:57 maps
```

Reply:
Oops, sorry about that!

Reply:
@czender, please try again when you have a chance.

Reply:
Thanks @xylar, that fixed it. I can now copy map-files to `/lcrc/group/e3sm/public_html/diagnostics/maps` where `mache` will look for them. @chengzhuzhang and @forsyth2 shall I copy them:
1. Incrementally (one-by-one) when asked?
2. Copy a group of what I think the most-used maps are now?
3. Wait for someone to gather a list of the maps potentially requestable by the `e2c` and `zppy` codes?
4. Something else?

Reply:
Hi @czender Would you start copy an initial group of maps which you think are most-used. And we will let you know if any maps are missing, so that you could help add more?

Reply:
OK, I copied the most relevant ne30pg2 and ne30np4 to CMIP map-files. I can transfer more incrementally on request.
```
ac.zender@blueslogin3:~$ ls -l /lcrc/group/e3sm/public_html/diagnostics/maps
total 2956609
-rw-rw-r-- 1 ac.xasay-davis E3SM  49713640 Oct 11 22:45 map_EC30to60E2r2_to_cmip6_180x360_aave.20221012.nc
-rw-rw-r-- 1 ac.xasay-davis E3SM  49736893 Oct 11 22:45 map_EC30to60E2r2_to_cmip6_180x360_mono.20221012.nc
-rw-rw-r-- 1 ac.xasay-davis E3SM  49725632 Oct 11 22:45 map_EC30to60E2r2_to_cmip6_180x360_nco.20221012.nc
-rw-rw-r-- 1 ac.xasay-davis E3SM  49949972 Oct 11 22:45 map_ECwISC30to60E2r1_to_cmip6_180x360_aave.20221012.nc
-rw-rw-r-- 1 ac.xasay-davis E3SM  49973225 Oct 11 22:45 map_ECwISC30to60E2r1_to_cmip6_180x360_mono.20221012.nc
-rw-rw-r-- 1 ac.xasay-davis E3SM  49961964 Oct 11 22:45 map_ECwISC30to60E2r1_to_cmip6_180x360_nco.20221012.nc
-rw-r--r-- 1 ac.zender      E3SM  16086627 Sep 10  2018 map_ne30np4_to_cmip6_180x360_aave.20181001.nc
-rw-r--r-- 1 ac.zender      E3SM  31652151 Nov 29  2019 map_ne30np4_to_cmip6_180x360_mono.20190601.nc
-rw-r--r-- 1 ac.zender      E3SM  16069020 Dec  3  2019 map_ne30np4_to_cmip6_180x360_nco.20190601.nc
-rw-r--r-- 1 ac.zender      E3SM  11140512 Jan 23  2020 map_ne30pg2_to_cmip6_180x360_aave.20200201.nc
-rw-r--r-- 1 ac.zender      E3SM  49851268 Jan 25  2020 map_ne30pg2_to_cmip6_180x360_mono.20200201.nc
-rw-r--r-- 1 ac.zender      E3SM  11149068 Sep  1  2020 map_ne30pg2_to_cmip6_180x360_nco.20200901.nc
-rw-rw-r-- 1 ac.xasay-davis E3SM 231161392 Oct 12 10:09 map_SOwISC12to60E2r4_to_cmip6_180x360_aave.20221012.nc
-rw-rw-r-- 1 ac.xasay-davis E3SM 231180245 Oct 12 10:09 map_SOwISC12to60E2r4_to_cmip6_180x360_mono.20221012.nc
-rw-rw-r-- 1 ac.xasay-davis E3SM 231173448 Oct 12 10:09 map_SOwISC12to60E2r4_to_cmip6_180x360_nco.20221012.nc
-rw-rw-r-- 1 ac.xasay-davis E3SM 231161392 Oct 12 10:09 map_SOwISC12to60E2r4_to_cmip6_720x1440_aave.20221012.nc
-rw-rw-r-- 1 ac.xasay-davis E3SM 231180245 Oct 12 10:09 map_SOwISC12to60E2r4_to_cmip6_720x1440_mono.20221012.nc
-rw-rw-r-- 1 ac.xasay-davis E3SM 231173448 Oct 12 10:09 map_SOwISC12to60E2r4_to_cmip6_720x1440_nco.20221012.nc
-rw-rw-r-- 1 ac.xasay-davis E3SM 207277892 Oct 12 10:08 map_WC14to60E2r3_to_cmip6_180x360_aave.20221012.nc
-rw-rw-r-- 1 ac.xasay-davis E3SM 207297641 Oct 12 10:08 map_WC14to60E2r3_to_cmip6_180x360_mono.20221012.nc
-rw-rw-r-- 1 ac.xasay-davis E3SM 207289948 Oct 12 10:08 map_WC14to60E2r3_to_cmip6_180x360_nco.20221012.nc
-rw-rw-r-- 1 ac.xasay-davis E3SM 207277892 Oct 12 10:08 map_WC14to60E2r3_to_cmip6_720x1440_aave.20221012.nc
-rw-rw-r-- 1 ac.xasay-davis E3SM 207297641 Oct 12 10:08 map_WC14to60E2r3_to_cmip6_720x1440_mono.20221012.nc
-rw-rw-r-- 1 ac.xasay-davis E3SM 207289948 Oct 12 10:08 map_WC14to60E2r3_to_cmip6_720x1440_nco.20221012.nc

```

Reply:
Thank you, Charlie!

Reply:
@czender, @chengzhuzhang I have synced the files over on LCRC with mache (most were already in e3sm/diagnostics as far as I can tell).

If you want me to sync to other machines tomorrow, let me know.  Otherwise, I'll assume you'll test on LCRC first and I can sync over once this PR goes in.